process/eventsHandler.go

```diff
@@ -57,78 +69,90 @@ func checkArgs(args ArgsEventsHandler) error {
-		)
-		eh.publisher.BroadcastChan() <- events
-	} else {
-		log.Info("received duplicated events for block",
-			"block hash", events.Hash,
-			"ignoring", true,
+			"will process", shouldProcessEvents,
 		)
+		eh.publisher.Broadcast(events)
+		return
 	}
+
+	log.Info("received duplicated events for block",
+		"block hash", events.Hash,
+		"processed", false,
+	)
 }
 
 // HandleRevertEvents will handle revents events received from observer
 func (eh *eventsHandler) HandleRevertEvents(revertBlock data.RevertBlock) {
 	shouldProcessRevert := true
 	if eh.config.CheckDuplicates {
 		revertKey := revertKeyPrefix + revertBlock.Hash
-		shouldProcessRevert = eh.tryCheckProcessedOrRetry(revertKey)
+		shouldProcessRevert = eh.tryCheckProcessedWithRetry(revertKey)
 	}
 
 	if shouldProcessRevert {
 		log.Info("received revert event for block",
 			"block hash", revertBlock.Hash,
-			"shouldProcess", shouldProcessRevert,
-		)
-		eh.publisher.BroadcastRevertChan() <- revertBlock
-	} else {
-		log.Info("received duplicated revert event for block",
-			"block hash", revertBlock.Hash,
-			"ignoring", true,
+			"will process", shouldProcessRevert,
 		)
+		eh.publisher.BroadcastRevert(revertBlock)
+		return
 	}
+
+	log.Info("received duplicated revert event for block",
+		"block hash", revertBlock.Hash,
+		"processed", false,
+	)
 }
 
 // HandleFinalizedEvents will handle finalized events received from observer
 func (eh *eventsHandler) HandleFinalizedEvents(finalizedBlock data.FinalizedBlock) {
 	shouldProcessFinalized := true
 	if eh.config.CheckDuplicates {
 		finalizedKey := finalizedKeyPrefix + finalizedBlock.Hash
-		shouldProcessFinalized = eh.tryCheckProcessedOrRetry(finalizedKey)
+		shouldProcessFinalized = eh.tryCheckProcessedWithRetry(finalizedKey)
 	}
 
 	if shouldProcessFinalized {
 		log.Info("received finalized events for block",
 			"block hash", finalizedBlock.Hash,
-			"shouldProcess", shouldProcessFinalized,
-		)
-		eh.publisher.BroadcastFinalizedChan() <- finalizedBlock
-	} else {
-		log.Info("received duplicated finalized event for block",
-			"block hash", finalizedBlock.Hash,
-			"ignoring", true,
+			"will process", shouldProcessFinalized,
 		)
+		eh.publisher.BroadcastFinalized(finalizedBlock)
+		return
 	}
+
+	log.Info("received duplicated finalized event for block",
+		"block hash", finalizedBlock.Hash,
+		"processed", false,
+	)
 }
 
-func (eh *eventsHandler) tryCheckProcessedOrRetry(blockHash string) bool {
+func (eh *eventsHandler) tryCheckProcessedWithRetry(blockHash string) bool {
 	var err error
 	var setSuccessful bool
 
+	numRetries := 0
 	for {
 		setSuccessful, err = eh.locker.IsEventProcessed(context.Background(), blockHash)
 
 		if err != nil {
 			if !eh.locker.HasConnection(context.Background()) {
 				log.Error("failure connecting to redis")
-				time.Sleep(time.Millisecond * setnxRetryMs)
+
+				if numRetries >= eh.maxLockerConRetries {
+					break
+				}
+
+				time.Sleep(retryDuration)
+				numRetries++
+
```

Comment:
maybe add some prints on L163 if err != nil or the setSuccessful is false in order to debug the future problems more easily.

Reply:
added more log messages